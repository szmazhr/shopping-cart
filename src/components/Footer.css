html {
  position: relative;
  min-height: 100%;
}

body {
  margin-bottom: 60px !important;
}

.footer {
  width: 100%;
  background-color: rgb(233, 233, 233);
  color: rgb(27, 27, 27);
  height: 60px;
  position: absolute;
  bottom: 0;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0 60px;
  font-size: 1.2rem;

  
}
.footer a {
  color: rgb(27, 27, 27);
}
.repo_link {
  display: flex;
  align-items: center;
  justify-content: center;
}
.repo_link a {
  text-decoration: none;
  display: contents;
}

.repo_link a img{
transition: filter 0.3s ease-in-out;
}
.repo_link a:hover img{
  filter: invert(88%) sepia(19%) saturate(841%) hue-rotate(96deg) brightness(98%) contrast(106%);
}