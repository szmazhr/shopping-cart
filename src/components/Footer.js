import './Footer.css';
import github from '../assets/github.svg';
import heart from '../assets/heart.svg';

function Footer() {
  return (
    <footer className="footer">
      <div>
        <span>
          Made with <img src={heart} alt="heart"></img> by {" "}
          <a
            href={`https://github.com/szmazhr`}
            target="_blank"
            rel="noreferrer"
          >
            Shahzar Mazhar
          </a>
        </span>
      </div>
      <div className="repo_link">
        <a href='https://github.com/szmazhr/shopping-cart' target="_blank" rel="noreferrer">
          <img src={github} alt="github" width={30} />
        </a>
      </div>
    </footer>
  );
}

export default Footer;