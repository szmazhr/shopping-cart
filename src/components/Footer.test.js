import { render, screen } from '@testing-library/react';
import Footer from './Footer';

describe('Footer', () => {
  it('renders correctly', () => {
    render(<Footer/>);

    expect(screen.getByRole('contentinfo')).toBeInTheDocument();
    expect(screen.getByText('Shahzar Mazhar')).toBeInTheDocument();
    expect(screen.getAllByRole('img').length).toBe(2); 
    expect(screen.getByRole('link', {name: 'github'})).toHaveAttribute('href', 'https://github.com/szmazhr/shopping-cart')
  });
  it('matches snapshot', () =>{
    const {asFragment} = render(<Footer/>);
    expect(asFragment()).toMatchSnapshot();
  })
});