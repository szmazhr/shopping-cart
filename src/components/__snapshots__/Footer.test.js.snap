// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches snapshot 1`] = `
<DocumentFragment>
  <footer
    class="footer"
  >
    <div>
      <span>
        Made with 
        <img
          alt="heart"
          src="heart.svg"
        />
         by  
        <a
          href="https://github.com/szmazhr"
          rel="noreferrer"
          target="_blank"
        >
          Shahzar Mazhar
        </a>
      </span>
    </div>
    <div
      class="repo_link"
    >
      <a
        href="https://github.com/szmazhr/shopping-cart"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="github"
          src="github.svg"
          width="30"
        />
      </a>
    </div>
  </footer>
</DocumentFragment>
`;
